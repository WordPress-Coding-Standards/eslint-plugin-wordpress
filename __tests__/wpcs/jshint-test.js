'use strict';

const eslint = require( 'eslint' );
const path = require( 'path' );
const stripAnsi = require( 'strip-ansi' );

const configFile = path.resolve( __dirname, '../../lib/configs/jshint.js' );
const linter = new eslint.CLIEngine({
	configFile,
	ignorePath: './.gitignore',
	useEslintrc: false,
});

describe( 'JSHint config tests', () => {
	let formatter;
	let report;

	beforeEach(() => {
		report = linter.executeOnFiles(['__tests__/wpcs/fixtures/*-fixture.js']);
		formatter = stripAnsi( linter.getFormatter( 'codeframe' ) );
	});

	test( 'should have correct error and warning counts', () => {
		expect( report.results ).toHaveLength( 2 );
		expect( report.errored ).toBeFalsy();
		expect( report.errorCount ).toBe( 9 );
		expect( report.warningCount ).toBe( 0 );
		expect( report.fixableErrorCount ).toBe( 1 );
		expect( report.fixableWarningCount ).toBe( 0 );
	});

	test( 'lint results should match snapshot', () => {
		let output = formatter( report.results );
		let { results } = report;
		for (let result of results) {
			let relativeFilePath = path.relative( __dirname, result.filePath );
			expect( stripAnsi( output ) ).toMatchSnapshot( relativeFilePath );
		}
	});
});
