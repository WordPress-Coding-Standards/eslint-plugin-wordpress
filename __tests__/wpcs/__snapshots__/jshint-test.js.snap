// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSHint config tests lint results should match snapshot: fixtures/semicolons-fixture.js 1`] = `
"error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/semicolons-fixture.js:3:1:
  1 | /* eslint-disable no-unused-vars */
  2 | var aa;
> 3 | var bb
    | ^
  4 | var cc ;
  5 | var dd;;
  6 | var ee; ;


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/semicolons-fixture.js:4:1:
  2 | var aa;
  3 | var bb
> 4 | var cc ;
    | ^
  5 | var dd;;
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/semicolons-fixture.js:5:1:
  3 | var bb
  4 | var cc ;
> 5 | var dd;;
    | ^
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/semicolons-fixture.js:6:1:
  4 | var cc ;
  5 | var dd;;
> 6 | var ee; ;
    | ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/strings-fixture.js:3:1:
  1 | /* eslint-disable no-unused-vars */
  2 | var aa = 'Strings should be contained in single quotes';
> 3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
    | ^
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/strings-fixture.js:4:1:
  2 | var aa = 'Strings should be contained in single quotes';
  3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
> 4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
    | ^
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
  7 | /* eslint-enable no-unused-vars */


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/strings-fixture.js:5:1:
  3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
> 5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
    | ^
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/strings-fixture.js:6:1:
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
> 6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
    | ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Strings must use singlequote (quotes) at __tests__/wpcs/fixtures/strings-fixture.js:6:10:
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
> 6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
    |          ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


9 errors found.
1 error potentially fixable with the \`--fix\` option."
`;

exports[`JSHint config tests lint results should match snapshot: fixtures/strings-fixture.js 1`] = `
"error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/semicolons-fixture.js:3:1:
  1 | /* eslint-disable no-unused-vars */
  2 | var aa;
> 3 | var bb
    | ^
  4 | var cc ;
  5 | var dd;;
  6 | var ee; ;


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/semicolons-fixture.js:4:1:
  2 | var aa;
  3 | var bb
> 4 | var cc ;
    | ^
  5 | var dd;;
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/semicolons-fixture.js:5:1:
  3 | var bb
  4 | var cc ;
> 5 | var dd;;
    | ^
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/semicolons-fixture.js:6:1:
  4 | var cc ;
  5 | var dd;;
> 6 | var ee; ;
    | ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/strings-fixture.js:3:1:
  1 | /* eslint-disable no-unused-vars */
  2 | var aa = 'Strings should be contained in single quotes';
> 3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
    | ^
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/strings-fixture.js:4:1:
  2 | var aa = 'Strings should be contained in single quotes';
  3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
> 4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
    | ^
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
  7 | /* eslint-enable no-unused-vars */


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/strings-fixture.js:5:1:
  3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
> 5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
    | ^
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Combine this with the previous 'var' statement (one-var) at __tests__/wpcs/fixtures/strings-fixture.js:6:1:
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
> 6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
    | ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Strings must use singlequote (quotes) at __tests__/wpcs/fixtures/strings-fixture.js:6:10:
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
> 6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
    |          ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


9 errors found.
1 error potentially fixable with the \`--fix\` option."
`;
